sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -1966,7 +1966,20 @@ case class ElementAt(left: Expression, right: Expression)
   }
 
   override def nullable: Boolean = left.dataType match {
-    case _: ArrayType => computeNullabilityFromArray(left, right)
+    case _: ArrayType =>
+      def specialNormalizeIndex: (Int, Int) => Int = {
+        (arrayLength: Int, index: Int) => {
+          if (index < 0) {
+            arrayLength + index
```

Comment:
this can still be negative and fail, right?

Reply:
calling nullable will not get exception or failed, if it's out of bounds, it's just returning a default true.

Reply:
yes, if the passing index is negative and arrayLength + index still < 0, it will still failed.

Reply:
do we need to cover the arrayLength + index still < 0 inside this specialNormalizeIndex ?